package-res/resources/web/pentaho/lang/ComplexChangeset.js

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  "./Base",
+  "../util/object",
+  "../util/error"
+], function(Base, O, error) {
+  "use strict";
+
+  /**
+   * @classDesc An `ComplexChangeset` represents a collection of `property` to `change` pairs.
+   *
+   * ### `ComplexChangeset` Property Key
+   * A `ComplexChangeset` property key is the name of a complex's `property` that had its value changed.
+   *
+   * ### `ComplexChangeset` Change Object
+   * A `ComplexChangeset` `change` object is the object representing the change made to a `property`.
+   *
+   * @name ComplexChangeset
+   * @memberOf pentaho.lang
+   *
+   * @amd pentaho/lang/ComplexChangeset
+   * @class
+   * @extends pentaho.lang.Base
+   */
+  return Base.extend("pentaho.lang.ComplexChangeset", /** @lend pentaho.lang.ComplexChangeset#*/{
+
+    /**
+     * Creates a `ComplexChangeset` with a given owner.
+     *
+     * @constructor
+     *
+     * @param {!pentaho.type.complex} owner - The complex where the change occurred.
+     */
+    constructor: function(owner) {
+      if(!owner) throw error.argRequired("owner");
+
+      this._owner = owner;
+      this._properties = {};
+    },
+
+    /**
+     * Gets the complex where the change occurred.
+     *
+     * @type !pentaho.lang.ComplexChangeset
+     * @readonly
+     */
+    get owner() {
+      return this._owner;
+    },
+
+    /**
+     * Determines if the `ComplexChangeset` contains the specified property.
+     *
+     * @param {string} property - The property name.
+     *
+     * @returns {!boolean} True if the `property` exists, false otherwise.
+     */
+    has: function(property) {
+      return O.hasOwn(this._properties, property);
+    },
+
+    /**
+     * Sets a new `property` to `change` pair if none exists,
+     * otherwise updates it.
+     *
+     * @param {!string} property - The property name.
+     * @param {any?} valueSpec - A object representing a property change.
+     */
+    set: function(property, valueSpec) {
```

Comment:
The documentation isn't right. Please check the original set method's parameters. Also, no need to have different argument names. Either use the same, or change both ;-)
Please add a TODO on the "does nothing comment".